lib/orderbook/OrderBook.ts

```diff
@@ -141,15 +140,22 @@ class OrderBook extends EventEmitter {
   }
 
   private updateOrderQuantity = (type: { [pairId: string]: Orders }, order: orders.StampedOrder, decreasedQuantity: number) => {
-    const object = this.getOrderMap(type, order);
-    object[order.id].quantity = object[order.id].quantity - decreasedQuantity;
-    if (object[order.id].quantity === 0) {
-      delete object[order.id];
+    const identifier = this.getIdentifier(order);
+
+    const orderMap = this.getOrderMap(type, order);
+    orderMap[order[identifier]].quantity = orderMap[order[identifier]].quantity - decreasedQuantity;
+    if (orderMap[order[identifier]].quantity === 0) {
+      delete orderMap[order[identifier]];
     }
   }
 
   private addOrder = (type: { [pairId: string]: Orders }, order: orders.StampedOrder) => {
-    this.getOrderMap(type, order)[order.id] = order;
+    const identifier = this.getIdentifier(order);
+    this.getOrderMap(type, order)[order[identifier]] = order;
+  }
+
+  private getIdentifier = (order: orders.StampedOrder): string => {
+    return ('internalId' in order) ? 'internalId' : 'id';
```

Comment:
Also, if we do keep this method I think it should return the identifier itself, not the name of the identifier.